packages/api/src/command/medical/patient/__tests__/store-query-init.test.ts

```diff
@@ -0,0 +1,99 @@
+import { PatientModel } from "../../../../models/medical/patient";
+import { mockStartTransaction } from "../../../../models/__tests__/transaction";
+import { makePatientModel } from "../../../../models/medical/__tests__/patient";
+
+import { storeQueryInit, StoreQueryParams } from "../query-init";
+import {
+  pdParams,
+  dqParams,
+  cqParams,
+  mockedPatientAllProgresses,
+  documentQueryProgress,
+  patientDiscovery,
+  consolidatedQuery,
+} from "./store-query-cmd";
+
+let patientModel: PatientModel;
+
+let patientModel_update: jest.SpyInstance;
+let patientModel_findOne: jest.SpyInstance;
+jest.mock("../../../../models/medical/patient");
+
+beforeEach(() => {
+  patientModel = makePatientModel();
+  mockStartTransaction();
+  patientModel_findOne = jest.spyOn(PatientModel, "findOne").mockResolvedValue(patientModel);
+  patientModel_update = jest.spyOn(patientModel, "update");
+});
+
+afterEach(() => {
+  jest.restoreAllMocks();
+});
+
+const checkPatientUpdateWith = (params: StoreQueryParams) => {
+  expect(patientModel_update).toHaveBeenCalledWith(
+    expect.objectContaining({
+      data: expect.objectContaining(params.cmd),
+    }),
+    expect.anything()
+  );
+};
+
+const checkPatientDataUpdatedWithMockedPatient = async (params: StoreQueryParams) => {
+  patientModel_findOne.mockResolvedValueOnce(mockedPatientAllProgresses);
+  patientModel_update = jest.spyOn(mockedPatientAllProgresses, "update");
+
+  await storeQueryInit(params);
+
+  checkPatientUpdateWith(params);
+};
+
+const checkUnchanged = (value: object) => {
+  expect(patientModel_update).toHaveBeenCalledWith(
+    expect.objectContaining({
+      data: expect.objectContaining(value),
+    }),
+    expect.anything()
+  );
+};
```

Comment:
This is not really checking if we didn't pass other params than the ones on `value`, right? This is what I understand this test expects this function to be doing: 

```
does not clear other patient data when running storeQueryInit with pdParams
```

Reply:
Its testing to make sure the previous values were not overwritten when updating

Reply:
So your assumption is correct